praw/models/reddit/emoji.py

```diff
@@ -89,6 +92,47 @@ def delete(self):
         )
         self._reddit.request("DELETE", url)
 
+    def update(
+        self,
+        mod_flair_only: Optional[bool] = None,
+        post_flair_allowed: Optional[bool] = None,
+        user_flair_allowed: Optional[bool] = None,
+    ):
+        """Update the permissions of an emoji in this subreddit.
+
+        :param mod_flair_only: (boolean) Indicate whether the emoji is
+            restricted to mod use only. Respects pre-existing settings if not
+            provided.
+        :param post_flair_allowed: (boolean) Indicate whether the emoji can be
+            used in post flair. Respects pre-existing settings if not provided.
+        :param user_flair_allowed: (boolean) Indicate whether the emoji can be
+            used in user flair. Respects pre-existing settings if not provided.
+
+        To restrict the emoji ``'test'`` in subreddit ``'testsub'`` to mod use
+        only, try:
+
+        .. code-block:: python
+
+           reddit.subreddit('testsub').emoji['test'].update(mod_flair_only=True)
+
+        """
+        data = {
+            "name": self.name,
+            "mod_flair_only": self.mod_flair_only,
+            "post_flair_allowed": self.post_flair_allowed,
+            "user_flair_allowed": self.user_flair_allowed,
+        }
+
+        if mod_flair_only is not None:
+            data["mod_flair_only"] = bool(mod_flair_only)
+        if post_flair_allowed is not None:
+            data["post_flair_allowed"] = bool(post_flair_allowed)
+        if user_flair_allowed is not None:
+            data["user_flair_allowed"] = bool(user_flair_allowed)
+
+        url = API_PATH["emoji_update"].format(subreddit=self.subreddit)
+        self._reddit.post(url, data=data)
```

Comment:
Does this `POST` request return anything? If so, can we return it from the method? It would be nice for tests if there were a way to confirm that the action succeeded.

Reply:
I didn't write this. I wouldn't know.

Reply:
I think the best option would be:

    other = self._reddit.post(url, data=data)  # I'm assuming this is an Emoji
    self.__dict__.update(other.__dict__)

which basically just mutates the current Emoji to have the new attributes from the updated version.

Reply:
Unfortunately, the return value, like add, provides nothing meaningful.

Reply:
Also unfortunately, the values aren't stable when `None` is provided. So we need to explicitly pass all the values accurately provide an update.